fineract-provider/src/main/java/org/apache/fineract/portfolio/delinquency/service/DelinquencyReadPlatformServiceImpl.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.portfolio.delinquency.service;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.apache.fineract.portfolio.delinquency.data.DelinquencyBucketData;
+import org.apache.fineract.portfolio.delinquency.data.DelinquencyRangeData;
+import org.apache.fineract.portfolio.delinquency.domain.DelinquencyRange;
+import org.apache.fineract.portfolio.delinquency.domain.DelinquencyRangeRepository;
+import org.apache.fineract.portfolio.delinquency.exception.DelinquencyBucketNotFoundException;
+import org.apache.fineract.portfolio.delinquency.mapper.DelinquencyRangeMapper;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class DelinquencyReadPlatformServiceImpl implements DelinquencyReadPlatformService {
+
+    private final JdbcTemplate jdbcTemplate;
+    private final DelinquencyRangeRepository repositoryRange;
+    private final DelinquencyRangeMapper mapperRange;
+
+    @Override
+    public Collection<DelinquencyRangeData> retrieveAllDelinquencyRanges() {
+        List<DelinquencyRange> delinquencyRangeList = repositoryRange.findAll();
+        return mapperRange.map(delinquencyRangeList);
+    }
+
+    @Override
+    public DelinquencyRangeData retrieveDelinquencyRange(Long delinquencyRangeId) {
+        DelinquencyRange delinquencyRangeList = repositoryRange.getReferenceById(delinquencyRangeId);
+        return mapperRange.map(delinquencyRangeList);
+    }
+
+    @Override
+    public Collection<DelinquencyBucketData> retrieveAllDelinquencyBuckets() {
+        final DelinquencyBucketMapper rm = new DelinquencyBucketMapper();
```

Comment:
Big no-no. Let's not introduce native queries for places where we don't need to. The only place we should use native queries is where things couldn't be solved by using JPQL/CriteriaAPI/etc.

This whole thing could be replaced by a simple Repository call with a Spring Data/JPQL Projection.

Reply:
Done!